src/app/personside/dialogpanel/Hurtigreferat/HurtigreferatContainer.tsx

```diff
@@ -0,0 +1,167 @@
+    isPosting,
+    PostResource
+} from '../../../../rest/utils/postResource';
+import {Meldingstype, SendMeldingRequest, Temagruppe} from '../../../../models/meldinger/meldinger';
+import {AppState} from '../../../../redux/reducers';
+import {sendMeldingActionCreator} from '../../../../redux/restReducers/sendMelding';
+import {AlertStripeFeil, AlertStripeSuksess} from 'nav-frontend-alertstriper';
+import Undertittel from 'nav-frontend-typografi/lib/undertittel';
+import {DeprecatedRestResource} from '../../../../redux/restReducers/deprecatedRestResource';
+import {PersonRespons} from '../../../../models/person/person';
+import {isLoadedPerson} from '../../../../redux/restReducers/personinformasjon';
+import {Select} from 'nav-frontend-skjema';
+import {getTemaFraCookie, setTemaCookie} from './temagruppeutils';
+
+interface StateProps {
+    sendMeldingResource: PostResource<SendMeldingRequest>;
+    person: DeprecatedRestResource<PersonRespons>;
+}
+
+interface DispatchProps {
+    sendMelding: (tekst: string, temaGruppe: string) => void;
+}
+
+type Props = StateProps & DispatchProps;
+
+const Style = styled.div`
+    ${theme.resetEkspanderbartPanelStyling};
+    filter: drop-shadow(0 0 1px rgba(0, 0, 0, 0.7));
+    label {
+        ${theme.visuallyHidden}
+    }
+`;
+
+const Padding = styled.div`
+    padding: 0 1rem;
+`;
+
+interface Tema {
+    kodeverk: string;
+    beskrivelse: string;
+}
+
+const temagruppeValg: Tema[] = [
+    {beskrivelse: 'Arbeid', kodeverk: Temagruppe.Arbeid},
+    {beskrivelse: 'Familie', kodeverk: Temagruppe.Familie},
+    {beskrivelse: 'Hjelpemiddel', kodeverk: Temagruppe.Hjelpemiddel},
+    {beskrivelse: 'Pensjon', kodeverk: Temagruppe.Pensjon},
+    {beskrivelse: 'Øvrig', kodeverk: Temagruppe.Øvrig}
+];
+
+function HurtigreferatContainer(props: Props) {
+    const initialTema = temagruppeValg.find(tema => tema.kodeverk === getTemaFraCookie());
+    const [valgtTema, setTemagruppe] = useState<Tema | undefined>(initialTema);
+    const [temaGruppeFeilmelding, setTemaGruppeFeilmelding] = useState(false);
+    const sendResource = props.sendMeldingResource;
+    console.log(valgtTema);
+
+    if (isFinishedPosting(sendResource)) {
+        return <AlertStripeSuksess>Meldingen ble sendt.</AlertStripeSuksess>;
+    }
+
+    if (isFailedPosting(sendResource)) {
+        return (
+            <AlertStripeFeil>Det skjedde en feil ved sending av melding: {sendResource.error.message}</AlertStripeFeil>
+        );
+    }
+
+    const sendMelding = (tekst: string) => {
+        if (!valgtTema) {
+            setTemaGruppeFeilmelding(true);
+            return;
+        }
+        if (isNotStartedPosting(props.sendMeldingResource)) {
+            props.sendMelding(tekst, valgtTema.kodeverk);
+        }
+    };
+
+    const velgTemagruppeHandler = (event: ChangeEvent<HTMLSelectElement>) => {
+        const tema = temagruppeValg.find(tema => tema.kodeverk === event.target.value);
+        setTemagruppe(tema);
```

Comment:
Uten å ha mentale kapasiteten til å ha oversikt, ser dette ut som man har "several source of truth"? At den ligger både i cookie og state? Kan det skje noe usynk her?

Reply:
Eller er de kanskje ikke ment å være helt i synk? Tror ikke jeg forstår dette 100%

Reply:
Mmm, du er kanskje innpå noe her.. Cookien her brukes for å hente inn tema fra forrige session. Kunne jo droppet lokal state og brukt cookie som en slags state, men det virker litt skittent.. usikker